### DISCUSSION TITLE
Ambient light sensor support

### DISCUSSION THREAD

**Discussion by ThatOneCalculator** [YaLTeR/niri#580] [Ideas]
Being able to change the screen brightness with an ambient light sensor would be great. GNOME has something like this built-in, and it works flawlessly out of the box. There are third party solutions, but they all try to do too much (i.e. turning the webcam into a faux sensor, trying to adjust gamma, time of day and geolocation, etc) and all end up failing in one way or another. If there's something in the compositor that could just watch `/sys/bus/iio/devices/iio:device0/in_illuminance_raw` (between 0 and 255) or even have a dbus listener for [`clightd`](https://github.com/FedeDP/Clightd) `busctl call org.clightd.clightd /org/clightd/clightd/Sensor/Als org.clightd.clightd.Sensor Caputure "sis" "" 5 ""`, that would be incredibly useful.

**Comment by YaLTeR** [comment]
Hm, well, maybe; though since currently there's no brightness control built into niri and you use other tools to set brightness anyway, it would probably be easier to just write a script for this.
